### DISCUSSION TITLE
Warn by mail or other mean when a particular computer connects

### DISCUSSION THREAD

**Discussion by PaliPalo** [Ylianst/MeshCentral#5482] [Ideas]
Lastly, some colleague's laptop had been stolen. So, I wandered if there is a meaning to be warned when the computer will connect to MeshCentral server. So that, I could take control via a terminal connection and, at least, remove anything that could lead to our organization's IT ; so then reduce the risk of a hack.

Hopefully, the computer is attached to a simple user and it has been disabled within our domain. But that does not warrant a 100% safe state, not even 99%...

I guess this may be set upon single computer, not all. Otherwise, we shall be warned anytime and, specifically concerned by GDPR (General Data Protection Regulation), this limit the rule's exception requirements to particular cases.

Moreover, I  think the usage of this setting should only be available to administrative accounts.

What do you think about this ability ? Should it be really useful and easy to implement ?

However, I did not find such feature within the actual version.

**Comment by si458** [comment]
this is already built in

you can enable the web notifications for the device by going to the device and clicking the pencil next to notifications,
(this will only display notifications on the webpage if you have the webpage open)
![Screenshot 2023-10-28 110247](https://github.com/Ylianst/MeshCentral/assets/765314/3d67902c-6426-4297-8616-aa9fc6dba7a2)

you can also enable messaging notifications, if you have any setup (pushover, telegram, etc) https://ylianst.github.io/MeshCentral/messaging/

you can also setup web push notifications (like chrome/firefox push alerts)
![Screenshot 2023-10-28 at 11-03-21 MefooCentral](https://github.com/Ylianst/MeshCentral/assets/765314/a534d43a-559c-47b1-ab76-b4f5f5f762a9)

you can also setup smtp to get email alerts (again will put link here when i find it)

also remember messaging alerts are sent every 60 seconds and joined together but the web push notitifications are sent instantly if i remember

hope this helps!

EDIT: also when the device connects you can check its external ip under details tab - Last agent address - which will help narrow down provider, location, etc (remember its only an ip address so its not physical proof for court!)

**Reply by PaliPalo** [reply]
Thanks for you precisions, Simon. I will take a look at the notification on device connection (I already saw the "all devices connection" setting that is not  what I want) moreover upon the Telegram (or others) push notifications.

**Reply by si458** [reply]
im not sure where you are seeing 'all devices connection' ?
but you can enable notifications for devices on per device, per whole group, whole server
hope this helps!
